### ISSUE TITLE
Wrong error handler causes Node.js process crashes

### ISSUE THREAD

**Issue by Tofull** [nodemailer/nodemailer#1124]
Hi there, 

I was trying to implement a simple function to send email through a OAuth2 2LO authentication. It worked very well. Then I put a wrong privateKey to test the error handler and my server brutally stopped.

### Example
Here is what I am expecting from my example: 
- Start the timer.
- In 3 seconds, it will get a promise which returns error.
- The timer should continue as expected.
- In 6 seconds, it will call nodemailer.createTransport.verify().
- This function should return a rejected promise as the `user`, `serviceClient` and `privateKey` of the transporter are wrong.
- The timer should continue as expected. (until Ctrl + C)

```js
// cat repeat_bug.js

'use strict';
Error.stackTraceLimit = Infinity;

const nodemailer = require("nodemailer");

const transporter = nodemailer.createTransport({
    host: "smtp.gmail.com",
    port: 465,
    secure: true,
    auth: {
        type: "OAuth2",
        user: "WRONG_DATA_TO_REPEAT_THE_BUG",
        serviceClient: "WRONG_DATA_TO_REPEAT_THE_BUG",
        privateKey: "WRONG_DATA_TO_REPEAT_THE_BUG",
    },
});

async function fake_error_generator(){
    return new Promise((resolve, reject) => {
        reject(new Error("Fake error"));
    });
};

async function good_error_handler(transporter) {
    try {
        await fake_error_generator().catch((error) => console.error('Good error handling', error));
    } catch (error) {
        console.error('No error detected as expected (async try catch behaviour...)', error)
    }
};

async function bad_error_handler(transporter) {
    try {
        await transporter.verify().catch((error) => console.error('Should catch a rejected promise, but no error detected here...', error));
    } catch (error) {
        console.error('No error detected as expected (async try catch behaviour...)', error)
    }
};

// Example
console.log('Starts the timer.');
setInterval(function () {console.log(new Date())}, 1000);
console.log('In 3 seconds, it will get a promise which returns error.');
console.log('And the timer should continue as expected.');
console.log();
console.log('In 6 seconds, it will call nodemailer.createTransport.verify().');
console.log('This function should return a rejected promise as the user, serviceClient and privateKey of the transporter are wrong.');
console.log('But the error kills the Node.js process... instead of being catched.');

setTimeout(() => {
    good_error_handler(transporter);
}, 3000);

setTimeout(() => {
    bad_error_handler(transporter);
}, 6000);

```

```log
$ node.exe repeat_bug.js


Starts the timer.
In 3 seconds, it will get a promise which returns error.
And the timer should continue as expected.

In 6 seconds, it will call nodemailer.createTransport.verify().
This function should return a rejected promise as the user, serviceClient and privateKey of the transporter are wrong.
But the error kills the Node.js process... instead of being catched.
2020-04-10T22:20:49.652Z
2020-04-10T22:20:50.652Z
2020-04-10T22:20:51.652Z
Good error handling Error: Fake error
    at Promise (./repeat_bug.js:20:16)
    at new Promise (<anonymous>)
    at fake_error_generator (./repeat_bug.js:19:12)
    at good_error_handler (./repeat_bug.js:26:15)
    at Timeout.setTimeout [as _onTimeout] (./repeat_bug.js:51:5)
    at ontimeout (timers.js:436:11)
    at tryOnTimeout (timers.js:300:5)
    at listOnTimeout (timers.js:263:5)
    at Timer.processTimers (timers.js:223:10)
2020-04-10T22:20:52.653Z
2020-04-10T22:20:53.653Z
2020-04-10T22:20:54.653Z
internal/crypto/sig.js:83
  var ret = this._handle.sign(key, passphrase, rsaPadding, pssSaltLength);
                         ^

Error: error:0906D06C:PEM routines:PEM_read_bio:no start line
    at Sign.sign (internal/crypto/sig.js:83:26)
    at XOAuth2.jwtSignRS256 (./node_modules/nodemailer/lib/xoauth2/index.js:368:14)
    at XOAuth2.generateToken (./node_modules/nodemailer/lib/xoauth2/index.js:167:30)
    at XOAuth2.getToken (./node_modules/nodemailer/lib/xoauth2/index.js:123:18)
    at SMTPConnection._handleXOauth2Token (./node_modules/nodemailer/lib/smtp-connection/index.js:1697:27)
    at SMTPConnection.login (./node_modules/nodemailer/lib/smtp-connection/index.js:540:22)
    at connection.connect (./node_modules/nodemailer/lib/smtp-transport/index.js:374:32)
    at SMTPConnection.once (./node_modules/nodemailer/lib/smtp-connection/index.js:215:17)
    at Object.onceWrapper (events.js:273:13)
    at SMTPConnection.emit (events.js:182:13)
    at SMTPConnection._actionEHLO (./node_modules/nodemailer/lib/smtp-connection/index.js:1313:14)
    at SMTPConnection._processResponse (./node_modules/nodemailer/lib/smtp-connection/index.js:942:20)
    at SMTPConnection._onData (./node_modules/nodemailer/lib/smtp-connection/index.js:749:14)
    at TLSSocket.SMTPConnection._onSocketData.chunk (./node_modules/nodemailer/lib/smtp-connection/index.js:195:44)
    at TLSSocket.emit (events.js:182:13)
    at addChunk (_stream_readable.js:283:12)
    at readableAddChunk (_stream_readable.js:264:11)
    at TLSSocket.Readable.push (_stream_readable.js:219:10)
    at TLSWrap.onStreamRead [as onread] (internal/stream_base_commons.js:94:17)

```

I also tried with the callback object instead of promise and always get my node.js killed.

### Related issues
This issue may be related to https://github.com/nodemailer/nodemailer/issues/615.

### Resolution
Replacing [nodemailer/lib/xoauth2/index.js:166](https://github.com/nodemailer/nodemailer/blob/master/lib/xoauth2/index.js#L166) : 

```js
let token = this.jwtSignRS256(tokenData);
```

by : 
```js
let token;
try {
    token = this.jwtSignRS256(tokenData);
} catch {
    return callback(new Error('Can\x27t generate token. Check your auth options'));
}
```
solved the issue in my case.

**Comment by harishkumarc** [comment]
We need a proper error handler here https://github.com/nodemailer/nodemailer/blob/c0345482024f412cbc801a014637f92bef938416/lib/xoauth2/index.js#L358

**Comment by andris9** [comment]
Fixed in nodemailer@6.4.7
